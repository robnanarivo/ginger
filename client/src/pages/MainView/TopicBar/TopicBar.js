import React from 'react';
import { alpha } from '@mui/material/styles';
import { useHistory, useParams } from 'react-router-dom';
import ListSubheader from '@mui/material/ListSubheader';
import List from '@mui/material/List';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

function displayTopics(listOfTopics, chosenTopic, updateListOfGroups) {
  const history = useHistory();
  const res = [];
  listOfTopics.forEach((topic) => {
    res.push(
      (topic.toLowerCase() === chosenTopic)
        ? (
          <ListItemButton
            key={topic}
            onClick={() => { history.push(`/${topic.toLowerCase()}`); updateListOfGroups(topic.toLowerCase()); }}
            sx={{ bgcolor: alpha('#E5E5E5', 0.3) }}
          >
            <ListItemText primary={topic.replace('_', ' ')} />
          </ListItemButton>
        )
        : (
          <ListItemButton
            key={topic}
            onClick={() => { history.push(`/${topic.toLowerCase()}`); updateListOfGroups(topic.toLowerCase()); }}
          >
            <ListItemText primary={topic.replace('_', ' ')} />
          </ListItemButton>
        )
      ,
    );
  });
  return res;
}

export default function TopicBar(props) {
  const { updateListOfGroups } = props;
  const { topic } = useParams();
  const listOfTopics = ['Joined', 'Suggestions', 'Public', 'Private', 'Local', 'Food', 'Entertainment', 'Sports'];
  return (
    <List
      sx={{
        width: '100%',
        maxWidth: 280,
        bgcolor: '#FFFFFF',
        maxHeight: 430,
        m: 8,
      }}
      component="nav"
      aria-labelledby="nested-list-subheader"
      subheader={
        (
          <ListSubheader component="div" id="nested-list-subheader" sx={{ bgcolor: '#E5E5E5' }}>
            Topics
          </ListSubheader>
        )
      }
    >
      {displayTopics(listOfTopics, topic, updateListOfGroups)}
    </List>
  );
}
