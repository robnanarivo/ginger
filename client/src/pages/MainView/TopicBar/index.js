import TopicBar from './TopicBar';

export default TopicBar;
