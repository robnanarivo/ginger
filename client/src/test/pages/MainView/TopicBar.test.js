import React from 'react';
import renderer from 'react-test-renderer';
import {
  screen,
  render,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import userEvent from '@testing-library/user-event';
import TopicBar from '../../../pages/MainView/TopicBar';
import 'jest-styled-components';

describe('TopicBar unit test', () => {
  it('TopicBar Snapshot test', async () => {
    const history = createMemoryHistory();
    const tree = renderer.create(
      <Router history={history}>
        <TopicBar />
      </Router>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Topic contains correct elements test', async () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <TopicBar />
      </Router>,
    );
    // contains public topic
    expect(screen.getByText('Public')).toBeInTheDocument();
    // contains All Groups Topic
    expect(screen.getByText('Joined')).toBeInTheDocument();
  });

  it('Topic Click test', async () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <TopicBar topic="Public" updateListOfGroups={() => {}} />
      </Router>,
    );
    // click All Groups topic
    const agButton = screen.getByText('Public');
    userEvent.click(agButton);
    expect(history.location.pathname).toBe('/public');
    // click Public topic
    const pubButton = screen.getByText('Private');
    userEvent.click(pubButton);
    expect(history.location.pathname).toBe('/private');
  });
});
