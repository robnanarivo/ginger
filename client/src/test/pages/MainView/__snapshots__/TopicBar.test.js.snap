// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicBar unit test TopicBar Snapshot test 1`] = `
<nav
  aria-labelledby="nested-list-subheader"
  className="MuiList-root MuiList-padding MuiList-subheader css-1ny30ma-MuiList-root"
>
  <div
    className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-ap3lwe-MuiListSubheader-root"
    id="nested-list-subheader"
  >
    Topics
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Joined
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Suggestions
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Public
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Private
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Local
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Food
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Entertainment
      </span>
    </div>
  </div>
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Sports
      </span>
    </div>
  </div>
</nav>
`;
